controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -275,15 +269,28 @@ export class FederatedGraphRepository {
       return undefined;
     }
 
-    // Composed schema version is not set when the federated graph was not composed.
+    const latestVersion = await this.db
```

Comment:
Can we avoid this duplication and maintain only one implementation that queries the latest valid schema?